### PR TITLE
initial draft of Azure_Functions.md

### PR DESCRIPTION
**Pull request by Amielll**

### CODE REVIEW

**Comment by Macdeini** [comment]
Looks good, maybe you could add some images to help with navigating the setup page

File: Topics/Development_Process/Azure_Functions.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+## Azure Setup
+
+Here we will go through everything we need to do in the [Azure Portal](https://portal.azure.com/).
+
+### Account Signup
+
+First, you will need an Azure account. If you don't have one, sign up for [Azure](https://azure.microsoft.com/).
+
+**Disclaimer**: You will need a credit card. If you are new to Azure, you are eligible for a free tier. Moreover, you can link your UofT email to your Azure account to be for further credit. Microsoft will only put a small holding fee to verify your identity.
```
Comment (Addison-Weatherhead) [inline Topics/Development_Process/Azure_Functions.md:29]: _Moreover, you can link your UofT email to your Azure account to be for further credit._

What's the benefit of using the UofT email? I don't totally get it. 

If there is some benefit, try making it more general, since non UofT folks will likely see and use this site. E.g. "You can link your institution email address to your account which will make X easier". 

If there is no benefit, I wouldn't mention UofT then.

Comment (Addison-Weatherhead) [reply Topics/Development_Process/Azure_Functions.md:29]: after re reading.. did you mean that the person gets more free credits? If so, I would rephrase to "Moreover, you can link your university email (if you have one) to your Azure account to get more free credits" or something

File: Topics/Development_Process/Azure_Functions.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+and add to your path:
+
+```bash
+echo 'export PATH="/opt/homebrew/opt/dotnet@6/bin:$PATH"' >> ~/.zshrc
+```
+
+Azure Functions can be developed in other languages like Python and JavaScript, so feel free to choose a different language if you'd like.
+
+_Optional_: If you are using VS Code, go ahead and install the Azure Functions extension as well. This extension (and other Azure extensions) work really well with VS Code.
```
Comment (Addison-Weatherhead) [inline Topics/Development_Process/Azure_Functions.md:110]: Great addition, extensions always make things way easier!

File: Topics/Development_Process/Azure_Functions.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+## Create an Azure Function
+
+Open up a terminal and `cd` into a directory you'd like to create our Azure Function project. Once there, run `func init`. Select `dotnet` followed by `c#`, and your Azure Functions project will be initialized.
```
Comment (Addison-Weatherhead) [inline Topics/Development_Process/Azure_Functions.md:114]: Small typo: Should be 

"into a directory you'd like to create _your_ Azure Function project"

File: Topics/Development_Process/Azure_Functions.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+- Open a browser and enter the url http://localhost:7071/api/funcHttp?name=MyName
+- Use a tool like `curl` or Postman to make a POST request to the server with a request body containing a "name".
+
+That's it! You've created your first Azure function.
+
+## Deploy an Azure Function
+
+First, change the `AzureWebJobsStorage` property inside `local.settings.json` to the value of the connection string you copied when creating the Storage account.
+
+Now, open up a terminal and run `az login`. This will open a browser page and prompt you to sign in to your Azure account.
```
Comment (Addison-Weatherhead) [inline Topics/Development_Process/Azure_Functions.md:177]: Maybe obvious to those more familiar, but since I'm not familiar with this.. what directory would one want to be in when running `az login`? In the location of the local.settings.json or the project directory? Or does it not matter?

**Review by Addison-Weatherhead — COMMENTED** [review]
Overall looks really great! Detailed and easy to follow. Just a couple small typos/clarifications then will be good to merge

File: Topics/Development_Process/Azure_Functions.md (hunk 1 above)
Comment (Amielll) [reply Topics/Development_Process/Azure_Functions.md:29]: Thank you for your suggestion, Addison!  Yes, linking your UofT email to your Azure account can grant you Azure credit, which will subsequently be used to pay for Azure services once your free tier is over.

I've gone ahead and updated the file to reflect this.

File: Topics/Development_Process/Azure_Functions.md (hunk 3 above)
Comment (Amielll) [reply Topics/Development_Process/Azure_Functions.md:114]: Good catch, fixed!

File: Topics/Development_Process/Azure_Functions.md (hunk 4 above)
Comment (Amielll) [reply Topics/Development_Process/Azure_Functions.md:177]: Not at all obvious if you've never used the CLI tool before!

You can run the `az login` command in any directory. You will be authenticated to run subsequent `az` commands in any directory now, such as deploying a resource, creating a new one tied to your subscription, etc.

I have updated the tutorial to make this more clear.
